methods/stringmethods.js

```diff
@@ -1,9 +1,134 @@
+String.prototype.ucFirst = function () {
+    let regexPattern = /^[a-z]/, resultOfCheck = regexPattern.test(this); //regex pattern, and test to match a-z character(lowercase)
+    let remainChar = this.slice(1); // remove first item from the string and return the rest string
+    if (resultOfCheck) {
+        return (this[0].toUpper()) + remainChar; //change case to uppercase
+    }
+    return (this[0] + remainChar);
+}
+
+// isQuestion, Return true if the string is a question (ending with a question mark)
+String.prototype.isQuestion = function () {
+    let regexPattern = /\b[\w\s]+\b[?]$/, resultOfCheck = regexPattern.test(this); // return true if the match the word with preceeding space
+    return resultOfCheck;
+}
+
+// listWord, Returns a list of the words in the string, as an Array.
+String.prototype.listWords = function () {
+    let regexPattern = /\b(\w|')+\b/gim, resultOfCheck = this.match(regexPattern); // return all match in an arrray
+    return resultOfCheck;
+}
+
+// wordCount, Returns the number of words in the string
+String.prototype.wordCount = function () {
+    let arrayOfWords = this.listWords(), wordCount = (arrayOfWords == null) ? 0 : arrayOfWords.length; // get the list of word in array, then return the length
+    return wordCount;
+}
+
+// toCurrency, Returns a currency representation of the String
+String.prototype.toCurrency = function () {
+    let regExPattern = /\d(?=(\d{3})+\.)/g; // regex pattern to match either 1decimal or more than one decimal place
+    let replacer = '$&,'; //add comma after a decimal number
+    return this.replace(regExPattern, replacer);  //replace the the string of number with the commaa separated number 
+}
+
+// fromCurrency, Returns a number representation of the Currency String
+String.prototype.fromCurrency = function () {
+    let regExPattern = /,/g; //pattern to match comma
+    return this.replace(regExPattern, ''); // replace comma with empty
+}
+
+// inverseCase method, Returns each letter in the string as an inverse of its current case
+String.prototype.inverseCase = function () {
+    let casedString = "";
+    for (let index = 0; index < this.length; index++) {
+        if (isUpperCase(this[index])) { // check if number is uppercase, then append it to casedString variable
+            let char = this[index].toLower();
+            casedString += char;
+            continue;
+        }
+        let char = this[index].toUpper(); // change an upper case character to lowercase
+        casedString += char;
+    }
+    return casedString;
+}
+
+// inverseCase, Returns the letters in alternating cases. It must start with a lower case
+String.prototype.alternatingCase = function () {
+    if (/[!]/g.test(this)) { // check if string contain special character then return error
+        return 'you test contain special character!';
+    }
+    let casedString = '';
+    for (let index = 0; index < this.length; index++) {
```

Comment:
The 'let' keyword is an ES6 syntax. Replace it with the corresponding ES5 syntax